src/controller/java/AndroidDeviceControllerWrapper.cpp

```diff
@@ -212,16 +213,23 @@ void AndroidDeviceControllerWrapper::SendNetworkCredentials(const char * ssid, c
     mCredentialsDelegate->SendNetworkCredentials(ssid, password);
 }
 
-void AndroidDeviceControllerWrapper::SendThreadCredentials(const chip::DeviceLayer::Internal::DeviceNetworkInfo & threadData)
+void AndroidDeviceControllerWrapper::SendThreadCredentials(const chip::DeviceLayer::Internal::ThreadOperationalDataset & threadData)
 {
     if (mCredentialsDelegate == nullptr)
     {
         ChipLogError(Controller, "No credential callback available to send Thread credentials.");
         return;
     }
 
-    ChipLogProgress(Controller, "Sending Thread credentials for channel %u, PAN ID %x...", threadData.ThreadChannel,
-                    threadData.ThreadPANId);
+    const chip::Thread::OperationalDataset & dataset = static_cast<const chip::Thread::OperationalDataset &>(threadData);
```

Comment:
Do you think we could create a new `chip::Thread::OperationalDataset` object with the contents of `threadData`? It could only be executed `#if CHIP_PROGRESS_LOGGING`. Or at least static_assert that both structures have equal sizes?

Reply:
The current `chip::Thread::OperationalDataset` extends from `chip::DeviceLayer::Internal::ThreadOperationalDataset`. There's no additional data fields in the subclass. Is it necessary to static_assert here?

Reply:
Yeah, I'm just worried that the assumption that there's no additional data may be (silently) violated at some point, so I think  `static_assert(sizeof(chip::Thread::OperationalDataset) == sizeof(chip::DeviceLayer::Internal::ThreadOperationalDataset))` is a way to express the assumption in code.